lib/libutee/tee_system_pta.c

```diff
@@ -76,3 +77,31 @@ TEE_Result tee_unmap(void *buf, size_t len)
 
 	return res;
 }
+
+TEE_Result tee_invoke_supp_plugin(const TEE_UUID *uuid, uint32_t cmd,
+				  uint32_t sub_cmd, void *buf, size_t len)
+{
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					       TEE_PARAM_TYPE_VALUE_INPUT,
+					       TEE_PARAM_TYPE_MEMREF_INOUT,
+					       TEE_PARAM_TYPE_NONE);
+	TEE_Param params[TEE_NUM_PARAMS] = { };
+
+	if (!uuid || (len && !buf))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	params[0].memref.buffer = (void *)uuid;
+	params[0].memref.size = sizeof(TEE_UUID);
+	params[1].value.a = cmd;
+	params[1].value.b = sub_cmd;
+	params[2].memref.buffer = buf;
+	params[2].memref.size = len;
+
+	res = invoke_system_pta(PTA_SYSTEM_SUPP_PLUGIN_INVOKE, param_types,
+				params);
+	if (res)
+		EMSG("Invoke tee-supplicant's plugin failed");
```

Comment:
I would make it a `DMSG()`:
```c
		DMSG("Invoke tee-supplicant's plugin failed: %#"PRIx32, res);
```

Reply:
Could you please explain why you want to use `DMSG()` instead `EMSG()`?
In this case we print an error message, that's why it seems `EMSG` is more suitable.